packages/core/src/telemetry/service.ts

```diff
@@ -16,8 +15,6 @@ import process from "process";
 import type { Options } from "@/config/options.js";
 import { getGitRemoteUrl } from "@/telemetry/remote.js";
 
-const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
```

Comment:
A quick search yielded this solution for ESM:
```
import { fileURLToPath } from "node:url";
import path from "node:path";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
```
I'm not sure where I found the `new URL(".", ...)` bit. Perhaps that was causing the issue/inconsistency.

My preference would be to use the file rather than the string to temp file approach if we can. Do you know in which scenarios this old code was working and not working?

Reply:
It seems to break after build & minification, which causes `telemetry/detached-flush.js` to no longer be in the same path as `telemetry/service.ts`. So calling `import.meta.url` technically returns the correct path for where it's being called from, but we can't rely on it because we can't guarantee the two files will always be in the same dir. 

Old code seems to be failing more times than not due to this.

The ideal solution would be using `import.meta.resolve('./detached-script.js')` (https://nodejs.org/api/esm.html#importmetaresolvespecifier) but only stable in node 20+.